backend/internal/service/handler/spotify/fetch_albums.go

```diff
@@ -0,0 +1,105 @@
+package spotify
+
+import (
+	"platnm/internal/models"
+	"platnm/internal/service/ctxt"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/zmb3/spotify/v2"
+)
+
+type AlbumArtist struct {
+	AlbumID  int
+	ArtistID int
+}
+
+type AddedContent struct {
+	Albums  []*models.Album
+	Artists []*models.Artist
+}
+
+func (h *SpotifyHandler) GetNewReleases(c *fiber.Ctx) error {
+	newReleases, err := fetchNewReleasesFromSpotify(c)
+	if err != nil {
+		return err
+	}
+
+	addedContent := &AddedContent{}
+	for _, album := range newReleases.Albums {
+		if err := h.handleAlbum(c, album, addedContent); err != nil {
+			return err
+		}
+	}
+
+	return c.Status(fiber.StatusOK).JSON(addedContent)
+}
+
+func (h *SpotifyHandler) handleAlbum(c *fiber.Ctx, album spotify.SimpleAlbum, addedContent *AddedContent) error {
+	albumId, err := h.MediaRepository.GetExistingAlbumBySpotifyID(c.Context(), album.ID.String())
+	if err != nil {
+		return err
+	}
+
+	if albumId != nil {
+		return nil
+	}
+
+	addedAlbum, err := h.MediaRepository.AddAlbum(c.Context(), &models.Album{
+		SpotifyID:   album.ID.String(),
+		Title:       album.Name,
+		ReleaseDate: album.ReleaseDateTime(),
+		Cover:       album.Images[0].URL,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	for _, artist := range album.Artists {
+		if err := h.handleArtist(c, artist, addedContent, addedAlbum.ID); err != nil {
+			return err
+		}
+	}
+
+	addedContent.Albums = append(addedContent.Albums, addedAlbum)
+	return nil
+}
+
+func (h *SpotifyHandler) handleArtist(c *fiber.Ctx, artist spotify.SimpleArtist, addedContent *AddedContent, albumId int) error {
```

Comment:
```suggestion
func (h *SpotifyHandler) handleArtist(ctx context.Context, artist spotify.SimpleArtist, addedContent *AddedContent, albumId int) error {
```
then change arg when called from `c` ->  `c.Context()`